### ISSUE TITLE
`pure` and `minify` does not remove function calls

### ISSUE THREAD

**Issue by ooflorent** [evanw/esbuild#1935]
I don't know if I missed something or if it is the intended way but it seems that `minify` does not remove `pure` calls when bundling.

Considering the following files:
```ts
// pure_assert.ts
export function assert(expr: unknown, message?: string): asserts expr {
  if (!expr) {
    throw new Error(message);
  }
}
```
```ts
// pure_test.ts
import { assert } from "./pure_assert";
export function test_pure() {
  assert(true);
  return true;
}
```
When compiled using this command:
```
esbuild pure_test.ts --bundle --format=esm --pure:assert --minify-syntax
```
The output is:
```ts
// pure_assert.ts
function assert(expr, message) {
  if (!expr)
    throw new Error(message);
}

// pure_test.ts
function test_pure() {
  return assert(!0), !0;
}
export {
  test_pure
};
```
The call to `assert` should have been removed.

**Comment by evanw** [comment]
The [documentation for `--pure`](https://esbuild.github.io/api/#pure) explicitly calls out that this only works for references to globals:

> You can also configure additional globals to be marked `/* @__PURE__ */` as well.

You are trying to use this for local variables, which doesn't work. It's restricted to globals because applying this to local variables using only a name would be very error-prone, since any library you use might randomly break because it happens to have a local variable with the same name.

To get the behavior you want using pure annotations, you'd need to annotate each call expression as `/* @__PURE__ */` instead.

A simpler option would be to just make sure the assert function body is empty. Then all calls will be automatically removed by esbuild when minifying (using a relatively recent feature that was released in [version 0.14.10](https://github.com/evanw/esbuild/releases/tag/v0.14.10)):

```ts
// pure_assert.ts
export function assert(expr: unknown, message?: string): asserts expr {
  if (DEBUG && !expr) {
    throw new Error(message);
  }
}
```

Building this with `esbuild pure_test.ts --bundle --format=esm --define:DEBUG=false --minify-syntax` gives the following output:

```js
// pure_test.ts
function test_pure() {
  return !0;
}
export {
  test_pure
};
```

**Comment by ooflorent** [comment]
Thanks for the quick response. I missed this part in the documentation. Indeed adding the annotation works as intended.
